cmake/Modules/Packages/KOKKOS.cmake

```diff
@@ -89,7 +91,11 @@ else()
   if(CMAKE_REQUEST_PIC)
     set(CMAKE_POSITION_INDEPENDENT_CODE ON)
   endif()
-  add_subdirectory(${LAMMPS_LIB_KOKKOS_SRC_DIR} ${LAMMPS_LIB_KOKKOS_BIN_DIR} EXCLUDE_FROM_ALL)
+  if(Kokkos_INSTALL_HEADER)
+    add_subdirectory(${LAMMPS_LIB_KOKKOS_SRC_DIR} ${LAMMPS_LIB_KOKKOS_BIN_DIR})
+  else()
+    add_subdirectory(${LAMMPS_LIB_KOKKOS_SRC_DIR} ${LAMMPS_LIB_KOKKOS_BIN_DIR} EXCLUDE_FROM_ALL)
+  endif()
```

Comment:
I think you should be able to write it all as one line with something like:
```
add_subdirectory(${LAMMPS_LIB_KOKKOS_SRC_DIR} ${LAMMPS_LIB_KOKKOS_BIN_DIR} $<$<NOT:$<BOOL:${Kokkos_INSTALL_HEADER}>>:EXCLUDE_FROM_ALL>)
```

Reply:
Sorry, `add_subdirectory()` do not support generator expressions, so nothing to be done here.